dbcon/execplan/functioncolumn.cpp

```diff
@@ -323,6 +324,50 @@ void FunctionColumn::unserialize(messageqcpp::ByteStream& b)
   Func_decode* decode = dynamic_cast<Func_decode*>(fFunctor);
   if (decode)
     fFunctor = fDynamicFunctor = new Func_decode();
+
+  Func_json_length* json_length = dynamic_cast<Func_json_length*>(fFunctor);
```

Comment:
I am curios about this design decision?
Could you ellaborate on why do you need this RTTI-based checks to allocate fDynamicFunctor and replace fFunctor?

Reply:
Purpose: reinitialize the path variable.
Well, As we know, the variable `fDynamicFunctor` is used to reinitialize member variables(e.g `fSeed1、fSeed2、fSeedSet` in `Func_rand`). These functions use these variable to record the states in a SQL. Since the function state may be different in each SQL.
So these function must be assigned to `fDynamicFunctor` to ensure that is destructed after the call, does not retain the state of last SQL.
`FunctionColumn::~FunctionColumn()
{
  if (fDynamicFunctor)
    delete fDynamicFunctor;
}`

Reply:
Same as json function. Some json function will accept path parameters. These functions use variable path to save path state. Taking `json_length` as example, When a SQL is called, json_length will be check if path argument is constant, If it is constant, it will be cached until next SQL call.Refer to [here](https://github.com/qggcs/mariadb-columnstore-engine/blob/dd95fcdde4ac5b66db8409aa0ead03e2f5a02f0f/utils/funcexp/func_json_length.cpp#L45-L64).

The cache must be reinitialized before the next call, otherwise, it will affect the next SQL call. So these function are assigned to the `fDynamicFunctor`.

[json_length](https://github.com/qggcs/server/blob/a5dc76a051ff908a5f506ed12e8d05311752e5fb/sql/item_jsonfunc.cc#L2721) in mariadb server.

Reply:
You can run the func_json_length.test before and after delete RTTI code.